src/app/pages/wallets/my-wallets/wallet-form-modal/wallet-form-modal.component.spec.ts

```diff
@@ -33,4 +34,36 @@ describe('WalletFormModalComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('Should display error message when invalid data is provided', () => {
+
+    it('Empty name', () => {
+      component.form.get('name')?.setValue('');
+
+      fixture.detectChanges();
+
+      const errorMessageDiv = fixture.debugElement.query(By.css('.mat-error')); // null :(
+      expect(errorMessageDiv.nativeElement.textContent).toBe('Name is required');
+    });
+
+    it('Name with more than 20 characters', () => {
+
+      component.form.setValue({
+        name: "asdasdsadasdasdasdasdasdasdasdsad",
+      })
+      //flushMicrotasks(); // fakeAsync tez nei dziala :(
+      fixture.detectChanges();
+
+      const errorMessageDiv = fixture.debugElement.query(By.css('.mat-error')); // null :(
```

Comment:
This is actually true - it is null correctly. Your method `showNameError` never returns true because this form control / field is not `touched`. After using `component.form.setValue` you can use `component.form.markAsTouched` and that way this condition will be true.